sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -291,7 +291,7 @@ public void testSetOperation() {
       unioned.collectAsList());
 
     Dataset<String> subtracted = ds.except(ds2);
-    Assert.assertEquals(Arrays.asList("abc", "abc"), subtracted.collectAsList());
+    Assert.assertEquals(Arrays.asList("abc"), subtracted.collectAsList());
```

Comment:
Except apparently does not remove duplicates:
```scala
range(0, 10).registerTempTable("a")
range(5, 15).registerTempTable("b")
sql("(select * from a union all select * from a) except select * from b")
```
results in:
```
+---+
| id|
+---+
|  2|
|  2|
|  0|
|  0|
|  3|
|  3|
|  4|
|  4|
|  1|
|  1|
+---+
```
So that is weird.